airbyte-integrations/connectors/source-snipeit/source_snipeit/streams.py

```diff
@@ -0,0 +1,283 @@
+        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
+
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        if self.stop_immediately:
+            return None
+        elif self.offset < self.total:
+            self.offset += self.limit_per_page
+            return {"offset": self.offset}
+        else:
+            return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        """
+        Usually contains common params e.g. pagination size etc.
+        """
+        if next_page_token:
+            return {'limit': self.limit_per_page, 'offset': next_page_token.get("offset", None)}
+        else:
+            return {'limit': self.limit_per_page, 'offset': self.offset}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        self.total = response.json().get("total", 0)
+        yield from response.json().get("rows", [])
+
+
+# NOTE: Temporary, dirty hack to account for the import differences
+#       between airbyte_cdk 0.1.53 and 0.1.55
+from airbyte_cdk.sources.streams import IncrementalMixin
+
+class Events(SnipeitStream, IncrementalMixin):
+    primary_key = "id"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._state = {}
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        self._state[self.cursor_field] = value
+
+    @property
+    def cursor_field(self):
+        return "updated_at"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # NOTE: This was originally the key function in the max() call below
+        #       I moved it out here to keep mypy happy.
+        def __key_function(item: Tuple) -> Any:
+            return item[1]
+        if current_stream_state == {}:
+            self.state = latest_record[self.cursor_field]
+            return {self.cursor_field: latest_record[self.cursor_field]}
+        else:
+            records = {}
+            records[current_stream_state[self.cursor_field]] = arrow.get(current_stream_state[self.cursor_field])
+            records[latest_record[self.cursor_field]] = arrow.get(latest_record[self.cursor_field])
```

Comment:
@cornjuliox can you check how other connectors implement this function? My guess you're overcomplicating stuff here. 
* First, try to use datetime lib instead of arrow.
* Second, the API has a contract where they says what is the format for your state; you can apply a datetime strp function to extract the value and throw an error if fails.